src/FiberTraces.cc

```diff
@@ -2775,6 +2775,50 @@ namespace pfsDRPStella = pfs::drp::stella;
       }
       return -1;
     }
+    
+    template< typename ImageT, typename MaskT, typename VarianceT, typename arrayT, typename ccdImageT, int dim >
+    bool addFiberTraceToCcdArray( FiberTrace< ImageT, MaskT, VarianceT > const& fiberTrace,
+                                  afwImage::Image< arrayT > const& fiberTraceRepresentation,
+                                  ndarray::Array< ccdImageT, 2, dim > & ccdArray ){
+
+      const PTR(const FiberTraceFunction) ftf = fiberTrace.getFiberTraceFunction();
+      const ndarray::Array< float, 1, 1 > xCenters = fiberTrace.getXCenters();
+      ndarray::Array< size_t, 2, 1 > minCenMax = pfsDRPStella::math::calcMinCenMax( xCenters,
+                                                                                    ftf->fiberTraceFunctionControl.xHigh,
+                                                                                    ftf->fiberTraceFunctionControl.xLow,
+                                                                                    ftf->fiberTraceFunctionControl.nPixCutLeft,
+                                                                                    ftf->fiberTraceFunctionControl.nPixCutRight);
+      /// Check fiberTraceRepresentation dimension
+      if ( fiberTraceRepresentation.getDimensions()[ 0 ] != ( minCenMax[ 0 ][ 2 ] - minCenMax[ 0 ][ 0 ] + 1 ) ){
+        string message("pfs::drp::stella::math::addFiberTraceToCcdImage: ERROR: fiberTraceRepresentation.getDimensions()[ 0 ](=");
+        message += to_string( fiberTraceRepresentation.getDimensions()[ 0 ] ) + " != ( minCenMax[ 0 ][ 2 ] - minCenMax[ 0 ][ 0 ] + 1 )(= ";
+        message += to_string( minCenMax[ 0 ][ 2 ] - minCenMax[ 0 ][ 0 ] + 1 ) + ")";
+        cout << message << endl;
```

Comment:
This is a clumsy way to set a string.

Why are you writing this string to `cout`?  If the exception is caught it's the catching code's responsibility.

Also, if you must print it use logging (and even if you didn't do that, you'd write to `cerr`).